flash/core/model.py

```diff
@@ -143,9 +145,16 @@ def predict(
         """
         running_stage = RunningStage.PREDICTING
         data_pipeline = data_pipeline or self.data_pipeline
+
+        # This check is necessary for mypy compatibility
+        if not isinstance(data_pipeline, DataPipeline):
+            raise TypeError(
+                f'data_pipeline should be of type DataPipeline, but is of type {type(data_pipeline).__name__}!'
+            )
+
         x = [x for x in data_pipeline._generate_auto_dataset(x, running_stage)]
         x = data_pipeline.worker_preprocessor(running_stage)(x)
-        x = self.transfer_batch_to_device(x, self.device)
+        x = self.transfer_batch_to_device(x, torch.device(self.device))
```

Comment:
I guess the best fix for this would be to add the correct type to Lightning.

Should we create a .pyi stub file instead for these?

Reply:
We could, I tried to solve it in flash only for now, to avoid touching lightning just for this.

Reply:
the .pyi stub file would live in Flash, until the typing is fixed in Lightning

lightning.pyi

```python
class LightningModule:
    device: torch.Device
```

Not an expert on this, not sure if this works. Copying examples from https://github.com/python/typeshed

Reply:
That's nice. didn't know one could have stub files for other libs included